src/main/java/io/pravega/perf/PravegaWriterWorker.java

```diff
@@ -16,31 +16,41 @@
 import io.pravega.client.ClientFactory;
 import io.pravega.client.stream.impl.ByteArraySerializer;
 import io.pravega.client.stream.EventWriterConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Class for Pravega writer/producer.
  */
 public class PravegaWriterWorker extends WriterWorker {
+    private static Logger log = LoggerFactory.getLogger(PravegaWriterWorker.class);
+
     final EventStreamWriter<byte[]> producer;
 
     PravegaWriterWorker(int sensorId, int events, int EventsPerFlush, int secondsToRun,
                         boolean isRandomKey, int messageSize, long start,
                         PerfStats stats, String streamName, int eventsPerSec,
-                        boolean writeAndRead, ClientFactory factory) {
+                        boolean writeAndRead, ClientFactory factory,
+                        boolean enableConnectionPooling) {
 
         super(sensorId, events, EventsPerFlush,
                 secondsToRun, isRandomKey, messageSize, start,
                 stats, streamName, eventsPerSec, writeAndRead);
 
+        log.info("enableConnectionPooling={}", enableConnectionPooling);
+
         this.producer = factory.createEventWriter(streamName,
                 new ByteArraySerializer(),
-                EventWriterConfig.builder().build());
+                EventWriterConfig.builder()
+                        .enableConnectionPooling(enableConnectionPooling)
+                        .build());
     }
 
     @Override
     public long recordWrite(byte[] data, TriConsumer record) {
         CompletableFuture ret;
         final long time = System.currentTimeMillis();
+        record.accept(time, -1, data.length);
```

Comment:
This will reduce the strength of the tool to pump more data to pravega client; this command method is  used kafka bench-marking too.
implement it as different method something like "recordWriteAck"  and invoke when the user specifies the "-unack true" .

Reply:
Hi @claudiofahey 
I have raised the pull request : https://github.com/pravega/pravega-benchmark/pull/61 
to extend the tool for kafka bench-marking;  if you are make any changes in the tool ; I suggest you test with kafka too;  This bench-marking tool and existing kafka benchmark tool should yield same results for the single producer.

Reply:
I will update the PR once #61 has been merged to master.